game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbTokenGenerator.java

```diff
@@ -0,0 +1,129 @@
+package games.strategy.engine.posted.game.pbf;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.framework.system.HttpProxy;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.snakeyaml.engine.v2.api.Load;
+import org.snakeyaml.engine.v2.api.LoadSettings;
+
+@AllArgsConstructor
+public class NodeBbTokenGenerator {
+  private final String forumUrl;
+  private final Load load = new Load(LoadSettings.builder().build());
+
+  @AllArgsConstructor(access = AccessLevel.PRIVATE)
+  @Getter
+  public static class TokenInfo {
+    @Nonnull private final String token;
+    private final int userId;
+  }
+
+  public TokenInfo generateToken(
+      final String username, final String password, @Nullable final String otp) {
+    Preconditions.checkNotNull(username);
+    Preconditions.checkNotNull(password);
+
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      final int userId = getUserId(client, username);
+      return new TokenInfo(getToken(client, userId, password, otp), userId);
+    } catch (final IOException e) {
+      throw new RuntimeException("Failed to retrieve login token", e);
+    }
+  }
+
+  public void revokeToken(final String token, final int userId) {
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      deleteToken(client, userId, token);
+    } catch (final IOException e) {
+      throw new RuntimeException("Failed to revoke login token", e);
+    }
+  }
+
+  private void deleteToken(final CloseableHttpClient client, final int userId, final String token)
+      throws IOException {
+    final HttpDelete httpDelete =
+        new HttpDelete(forumUrl + "/api/v2/users/" + userId + "/tokens/" + token);
+    HttpProxy.addProxy(httpDelete);
+    httpDelete.addHeader("Authorization", "Bearer " + token);
+    client.execute(httpDelete).close(); // ignore errors, execute and then close
```

Comment:
ignore error is a bit scary :grin: 

Should we at least info log them? Have you seen this throw errors?

Reply:
This is super funny, because you introduced this change in the first place ^^

Took me a while to find the commit though: https://github.com/triplea-game/triplea/pull/3715/files#diff-317b30fefdd065ba13162cb51844ab22L93-R96